task-requests/script.js

```diff
@@ -109,6 +122,60 @@ const changeFilter = () => {
   taskRequestContainer.innerHTML = '';
 };
 
+function handleBadgeDeletion(badgeType, badgeContent) {
+  let newFilterType;
+  if (Array.isArray(filterStates[badgeType])) {
+    newFilterType = filterStates[badgeType].filter(
+      (type) => type !== badgeContent,
+    );
+  } else if (filterStates[badgeType] === badgeContent) {
+    newFilterType = '';
+  }
+  updateFilterStates(badgeType, newFilterType);
+  const checkboxValue =
+    badgeType === 'requestType' ? badgeContent : badgeContent.toUpperCase();
+  uncheckCheckbox(checkboxValue);
+}
+
+function deleteBadge(e, badgeType) {
+  const badgeContent = e.target.textContent;
+  handleBadgeDeletion(badgeType, badgeContent);
+}
+
+function showBadges() {
+  badgesContainer.innerHTML = '';
+  const statuses = filterStates?.status;
+  const requestTypes = filterStates?.requestType;
+
+  if (statuses) {
+    if (Array.isArray(statuses)) {
+      statuses.forEach((status) => {
+        const badge = createBadge(status, 'status');
+        badgesContainer.appendChild(badge);
+      });
+    } else if (statuses) {
+      const badge = createBadge(statuses, 'status');
+      badgesContainer.appendChild(badge);
+    }
+  }
+
+  if (requestTypes && requestTypes.length > 0) {
+    requestTypes.forEach((requestType) => {
+      const badge = createBadge(requestType, 'requestType');
+      badgesContainer.appendChild(badge);
+    });
+  }
+
+  if (
+    (statuses && statuses.length > 0) ||
+    (requestTypes && requestTypes.length > 0)
+  ) {
+    filtersHeader.style.display = 'flex';
+  } else {
+    filtersHeader.style.display = 'none';
+  }
+}
+
```

Comment:
`handleBadgeDeletion` function could be simplified by using a ternary operator to handle the logic for updating the filter state based on whether the badge type is an array or a single value.

Reply:
If we use ternary operator then it will turn into nested ternary like this 

```
const newFilterType = Array.isArray(filterStates[badgeType])
  ? filterStates[badgeType].filter((type) => type !== badgeContent)
  : (filterStates[badgeType] === badgeContent ? '' : filterStates[badgeType]);
```


I feel the current if else statements provide better readability then nested ternary. Please let me know if you have any other suggestions or concerns.

Reply:
Okay cool, since you find the current if-else statements provide better readability, we can stick to it for now.